### PR TITLE
Tests for provider and subscriber created

### PR DESCRIPTION
**Pull request by Chad-Petersen**
This code tries to resolve the provider and subscriber missing tests issues. And also ensuring that the identity is a string and not integer.

### CODE REVIEW

**Comment by allistercsmith** [comment]
A side note on the development environment in general - I rate we need to be using Docker compose here.

Currently, if you make any changes during develop, you need to restart the container. Rather, we should be using volumes to mount the local directory to the container, such that changes are reflected straight away.

I can do this in another PR perhaps?

File: src/Service/Provider.php [hunk 1]
```diff
@@ -35,20 +36,19 @@ class Provider implements ProviderInterface
      * A provider end point
      *
      * @access private
-     * @var    string A provider endpoint
+     * @var    Object\EndPointInterface A provider endpoint
```
Comment (allistercsmith) [inline src/Service/Provider.php]: Perhaps more of a personal preference, but I think using the full namespace will reduce the number of WTF's per second. I.e. Simio\Zappi\Object\EndPointInterface

File: src/Service/Provider.php [hunk 2]
```diff
@@ -59,35 +59,35 @@ public function __construct($identity, $name, $endPoint)
...
     {
-        if (!is_integer($identity) && $identity != null) {
-            throw new Exception\InvalidArgumentException(
-                sprintf(
-                    '%s expects a integer|null argument; received "%s"',
-                    __METHOD__,
-                    is_object($identity) ? get_class($identity) : gettype($identity)
-                )
-            );
+        if ($identity === null) {
```
Comment (allistercsmith) [inline src/Service/Provider.php]: I'd prefer being consistent with using PHP functions. I.e. `is_null($identity)`

File: src/Service/Provider.php [hunk 3]
```diff
@@ -117,19 +117,19 @@ public function hasIdentity()
      * @return void
      * @throws Exception\InvalidArgumentException if invalid arguments supplied
      */
-    public function setProviderName($name = "")
+    public function setName($name = "")
```
Comment (allistercsmith) [inline src/Service/Provider.php]: I don't think we need a default value for $name

File: src/Service/Provider.php [hunk 4]
```diff
@@ -117,19 +117,19 @@ public function hasIdentity()
      * @return void
      * @throws Exception\InvalidArgumentException if invalid arguments supplied
```
Comment (allistercsmith) [inline src/Service/Provider.php]: Same note about using full namespace applies

File: src/Service/Provider.php [hunk 5]
```diff
@@ -147,17 +147,17 @@ public function getProviderName()
      * The purpose of this method is to set the provider end point
      *
      * @access public
-     * @param  EndPointInterface $endPoint A provider end point
+     * @param  Object\EndPointInterface $endPoint A provider end point
```
Comment (allistercsmith) [inline src/Service/Provider.php]: Same note about full namespace

File: src/Service/Provider.php [hunk 6]
```diff
@@ -172,7 +172,7 @@ public function setProviderEndpoint(EndPointInterface $endPoint)
      * @access public
      * @return EndPointInterface
```
Comment (allistercsmith) [inline src/Service/Provider.php]: Need to be consistent with namespacing. Same note about full namespace applies, but even if we don't go that route, this needs to be `Object\EndPointInterface` to keep consistency.

File: src/Service/ProviderInterface.php [hunk 7]
```diff
@@ -47,30 +47,30 @@ public function hasIdentity();
      * @param  string $name A provider name
      * @return void
      */
-    public function setProviderName($name = "");
+    public function setName($name = "");
```
Comment (allistercsmith) [inline src/Service/ProviderInterface.php]: As mentioned above, don't see why we need a default value.

File: test/Service/ProviderTest.php [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+namespace SimioTest\Zappi\Service;
+
+use Simio\Zappi\Service\Provider;
+
+/**
+ * This is the Provider Test
+ *
+ * @category  Service
+ * @link      http://github.com/iosim/simio-zappi for the canonical source repository
+ * @copyright Copyright (c) 2015-2016 Simio Inc. (http://www.sim.io)
```
Comment (allistercsmith) [inline test/Service/ProviderTest.php]: Do we even own the domain yet? :P

**Comment by Chad-Petersen** [comment]
I second the use of docker-compose.

File: test/Service/ProviderTest.php (hunk 8 above)
Comment (Chad-Petersen) [inline test/Service/ProviderTest.php]: Nope hahaha -need to get one :)

File: src/Service/Provider.php (hunk 2 above)
Comment (Chad-Petersen) [inline src/Service/Provider.php]: Used is_null instead.

File: src/Service/Provider.php (hunk 3 above)
Comment (Chad-Petersen) [inline src/Service/Provider.php]: Removed empty string default.

**Pull request merged**